ppl/src/test/java/org/opensearch/sql/ppl/calcite/CalcitePPLAppendTest.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.ppl.calcite;
+
+import java.util.Arrays;
+import java.util.List;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.test.CalciteAssert;
+import org.junit.Test;
+
+public class CalcitePPLAppendTest extends CalcitePPLAbstractTest {
+
+  public CalcitePPLAppendTest() {
+    super(CalciteAssert.SchemaSpec.SCOTT_WITH_TEMPORAL);
+  }
+
+  @Test
+  public void testAppend() {
+    String ppl = "source=EMP | append [ source=EMP | where DEPTNO = 20 ]";
+    RelNode root = getRelNode(ppl);
+    String expectedLogical =
+        "LogicalUnion(all=[true])\n"
+            + "  LogicalTableScan(table=[[scott, EMP]])\n"
+            + "  LogicalFilter(condition=[=($7, 20)])\n"
+            + "    LogicalTableScan(table=[[scott, EMP]])\n";
+    verifyLogical(root, expectedLogical);
+    verifyResultCount(root, 19); // 14 original table rows + 5 filtered subquery rows
+
+    String expectedSparkSql =
+        "SELECT *\n"
+            + "FROM `scott`.`EMP`\n"
+            + "UNION ALL\n"
+            + "SELECT *\n"
+            + "FROM `scott`.`EMP`\n"
+            + "WHERE `DEPTNO` = 20";
+    verifyPPLToSparkSQL(root, expectedSparkSql);
+  }
+
+  @Test
+  public void testAppendEmptySearchCommand() {
+    List<String> testPPLs =
+        Arrays.asList("source=EMP | append [ | where DEPTNO = 20 ]", "source=EMP | append [ ]");
```

Comment:
Supporting `append [ ]` is on purpose?

Reply:
Get it, please support the corner case `append [ ]` and `append [ | fields a,b,c ]`